apps/emqx_bridge/src/emqx_bridge_v2.erl

```diff
@@ -1399,72 +1409,49 @@ bridge_v1_remove(
+    BridgeV2Type = ?MODULE:bridge_v1_type_to_bridge_v2_type(BridgeV1Type),
+    case emqx_bridge_lib:maybe_withdraw_rule_action(BridgeV2Type, BridgeName, RemoveDeps) of
         ok ->
-            case remove(BridgeType, BridgeName) of
-                ok when RemoveConnector ->
-                    maybe_delete_channels(BridgeType, BridgeName, ConnectorName);
+            #{connector := ConnectorName} = lookup_conf(BridgeV2Type, BridgeName),
+            case remove(BridgeV2Type, BridgeName) of
                 ok ->
-                    ok;
+                    ConnectorType = connector_type(BridgeV2Type),
+                    case
+                        lists:member(connector, RemoveDeps) andalso
+                            not connector_has_channels(ConnectorType, ConnectorName)
+                    of
+                        true -> delete_connector(ConnectorType, ConnectorName);
+                        false -> ok
+                    end;
                 {error, Reason} ->
                     {error, Reason}
             end;
         {error, Reason} ->
             {error, Reason}
-    end;
-bridge_v1_check_deps_and_remove(_BridgeType, _BridgeName, _RemoveDeps, Error) ->
-    %% TODO: the connector is gone, for whatever reason, maybe call remove/2 anyway?
-    Error.
+    end.
 
-maybe_delete_channels(BridgeType, BridgeName, ConnectorName) ->
-    case connector_has_channels(BridgeType, ConnectorName) of
-        true ->
+delete_connector(ConnectorType, ConnectorName) ->
+    case emqx_connector:remove(ConnectorType, ConnectorName) of
+        ok ->
             ok;
-        false ->
-            ConnectorType = connector_type(BridgeType),
-            case emqx_connector:remove(ConnectorType, ConnectorName) of
-                ok ->
-                    ok;
-                {error, Reason} ->
-                    ?SLOG(error, #{
-                        msg => failed_to_delete_connector,
-                        bridge_type => BridgeType,
-                        bridge_name => BridgeName,
-                        connector_name => ConnectorName,
-                        reason => Reason
-                    }),
-                    {error, Reason}
-            end
+        {error, Reason} ->
+            ?SLOG(error, #{
+                msg => failed_to_delete_connector,
+                connector_type => ConnectorType,
+                connector_name => ConnectorName,
+                reason => Reason
+            }),
+            {error, Reason}
     end.
 
-connector_has_channels(BridgeV2Type, ConnectorName) ->
-    ConnectorType = connector_type(BridgeV2Type),
-    case emqx_connector_resource:get_channels(ConnectorType, ConnectorName) of
-        {ok, []} ->
-            false;
-        _ ->
-            true
-    end.
+connector_has_channels(ConnectorType, ConnectorName) ->
+    {ok, Channels} = emqx_connector_resource:get_channels(ConnectorType, ConnectorName),
+    length(Channels) =/= 0.
```

Comment:
```suggestion
    case emqx_connector_resource:get_channels(ConnectorType, ConnectorName) of
        {ok, [_ | _]} -> true;
        _ -> false
    end.
```
?

Reply:
That would hide an unexpected error

Reply:
The idea for the suggestion is: if the channel doesn't exist, it doesn't have channels.

Reply:
you mean the connector... hm... I will change even though personally I'm not a big fan of such fuzziness.

Reply:
Oops, yes, the connector.  :speak_no_evil: 

I also like strictness.  But I think that, in this case, it's semantically correct to return false if the connector doesn't exist.

Reply:
An action pointing to a non-existent connector, sure, shouldn't be a valid state, but I'm just focusing on this specific function.